examples/advanced_operations/add_gmail_ad.rb

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example adds a Gmail ad to a given ad group. The ad group's campaign
+# needs to have an AdvertisingChannelType of DISPLAY and
+# AdvertisingChannelSubType of DISPLAY_GMAIL_AD. To get ad groups, run
+# get_ad_groups.rb.
+
+require 'optparse'
+require 'date'
+require 'open-uri'
+require 'google/ads/google_ads'
+require 'google/ads/google_ads/v1/errors/errors_pb'
+
+def content_type_to_symbol(content_type)
```

Comment:
Move this method down by the other one (get_image). Is this definitely a safe way to do this for any valid content type? If not we may want to do a more annoying manual lookup so we can raise an error if we get an unsupported content type. Users often actually re-use example code directly.

Reply:
this will work for all types that are supported by our protos, The other thing we could do is assert that the HTTP response gives us the response type we're expecting and raise if it doesn't. WDYT?

Reply:
That sounds good.